File: app/src/main/java/org/qp/android/helpers/adapters/AutoScrollRunnable.java
package org.qp.android.helpers.adapters;

import androidx.viewpager2.widget.ViewPager2;

public record AutoScrollRunnable(ViewPager2 viewPager ,
                                 int delayMillis ,
                                 boolean reverse) implements Runnable {
    @Override
    public void run() {
        var currentItem = viewPager.getCurrentItem();
        if (viewPager.getAdapter() == null) return;
        var itemCount = viewPager.getAdapter().getItemCount();
        var nextItem = reverse ? currentItem - 1 : currentItem + 1;
        if (nextItem >= itemCount) {
            nextItem = 0;
        } else if (nextItem < 0) {
            nextItem = itemCount - 1;
        }
        viewPager.setCurrentItem(nextItem , true);
        viewPager.postDelayed(this , delayMillis);
    }
}

File: app/src/main/java/org/qp/android/helpers/adapters/AutoScrollRunnable.kt
package org.qp.android.helpers.adapters

import androidx.viewpager2.widget.ViewPager2

class AutoScrollRunnable(
    private val viewPager: ViewPager2,
    private val delayMillis: Int,
    private val reverse: Boolean
) : Runnable {

    override fun run() {
        val currentItem = viewPager.currentItem
        if (viewPager.adapter != null) {
            val itemCount = viewPager.adapter!!.itemCount
            var nextItem = if (reverse) currentItem - 1 else currentItem + 1
            if (nextItem >= itemCount) {
                nextItem = 0
            } else if (nextItem < 0) {
                nextItem = itemCount - 1
            }
            viewPager.setCurrentItem(nextItem, true)
            viewPager.postDelayed(this, delayMillis.toLong())
        }
    }
}